internal/provider/resource_tfe_organization_run_task_test.go

```diff
@@ -184,6 +187,66 @@ func TestAccTFEOrganizationRunTask_Read(t *testing.T) {
 	})
 }
 
+func TestAccTFEOrganizationRunTask_HMACWriteOnly(t *testing.T) {
+	skipUnlessRunTasksDefined(t)
+
+	tfeClient, err := getClientUsingEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	org, orgCleanup := createBusinessOrganization(t, tfeClient)
+	t.Cleanup(orgCleanup)
+
+	runTask := &tfe.RunTask{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	// Create the value comparer so we can add state values to it during the test steps
+	compareValuesDiffer := statecheck.CompareValue(compare.ValuesDiffer())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFEOrganizationRunTaskDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFEOrganizationRunTask_hmacAndHMACWriteOnly(org.Name, rInt, runTasksURL()),
+				ExpectError: regexp.MustCompile(`Attribute "hmac_key_wo" cannot be specified when "hmac_key" is specified`),
+			},
+			{
+				Config: testAccTFEOrganizationRunTask_hmacWriteOnly(org.Name, rInt, runTasksURL(), "foo"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEOrganizationRunTaskExists("tfe_organization_run_task.foobar", runTask),
+					resource.TestCheckResourceAttr("tfe_organization_run_task.foobar", "hmac_key", ""),
+					resource.TestCheckNoResourceAttr("tfe_organization_run_task.foobar", "hmac_key_wo"),
+				),
+				// Register the id with the value comparer so we can assert that the
+				// resource has been replaced in the next step.
+				ConfigStateChecks: []statecheck.StateCheck{
+					compareValuesDiffer.AddStateValue(
+						"tfe_organization_run_task.foobar", tfjsonpath.New("id"),
+					),
+				},
+			},
+			{
+				Config: testAccTFEOrganizationRunTask_hmacWriteOnly(org.Name, rInt, runTasksURL(), "foo2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEOrganizationRunTaskExists("tfe_organization_run_task.foobar", runTask),
+					resource.TestCheckResourceAttr("tfe_organization_run_task.foobar", "hmac_key", ""),
+					resource.TestCheckNoResourceAttr("tfe_organization_run_task.foobar", "hmac_key_wo"),
+				),
+				// Ensure that the resource has been replaced
+				ConfigStateChecks: []statecheck.StateCheck{
+					compareValuesDiffer.AddStateValue(
+						"tfe_organization_run_task.foobar", tfjsonpath.New("id"),
+					),
+				},
+			},
+		},
```

Comment:
There's one other test step we should add here - it's when the user switched from the write-only value to the regular one.  That's another case where we expect the resource to be recreated. Should be easy enough to reuse the `testAccTFEOrganizationRunTask_basic` config for this one.